src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -632,9 +632,15 @@ public function text(string $default = null, bool $normalizeWhitespace = true)
     /**
      * Returns only the inner text that is the direct descendent of the current node, excluding any child nodes.
      */
-    public function innerText(): string
+    public function innerText(bool $normalizeWhitespace = true): string
```

Comment:
Adding an argument is a BC break and should be done a bit differently:

```suggestion
    public function innerText(/* bool $normalizeWhitespace = true */): string
```

Then, add a phpdoc to document the new argument:

```
/**
 * @param bool $normalizeWhitespace
 */
```

Finally, use `func_num_args` and `func_get_arg` to retrieve the argument:

```
$normalizeWhitespace = 1 <= \func_num_args() ? func_get_arg(0) : false;
```

Note that I'm using `false` by default to keep the current behavior in 6.3.

Last, if someone does not pass the argument, raise a deprecation warning that the default value in 7.0 will change to `true`. So, if needed, people need to explicitly pass `false` to keep the current behavior in 7.0.

Reply:
About the default: the previous code was doing `$this->filterXPath('.//text()')->text()`, where `->text()` is the very method on this class, which defaults to `$normalizeWhitespace = true`, so it's fine using `true` as default for this method also.

Reply:
Thanks. I also just wanted to comment that. I think the new argument just offers the option to set it to `false` which isn't breaking backwards compatibility I guess.

Reply:
@nicolas-grekas You're right
@otsch Still for 6.3 anyway ;)